fsspec/implementations/sftp.py

```diff
@@ -13,6 +13,35 @@
 logger = logging.getLogger("fsspec.sftp")
 
 
+def _patch_SFTPFile(file):
```

Comment:
Hah, ugly :( (not a criticism of your work, just a shame)
Does this create reference cycles on `self` and the closure?

Other options might be:
- patch the upstream ftp.open to return a subclass `FsspecSShFile(SFTPFile, IOBase)`
- make our own class with the upstream file object as an attribute and pass all the methods through except for these two

Do these sound more palatable?

Reply:
Yeah, I kinda agree that it is not the most beautiful. I also considered a separate file object class, but I didn't want to copy 10+ methods with the danger of missing some. Subclassing also does not work because the object is created inside SFTPClient.open.

I have also opened a PR to fix this [upstream](https://github.com/paramiko/paramiko/pull/2456). I guess it is fine to wait for that to be merged and simply close this PR and issue here. I might have been too impatient. Fixing this here has the advantage that it will work with older versions of paramiko just fine, though.

Reply:
Ah, I'm glad you made that PR. In that case let's wait - it doesn't appear that this part of the API gets touched much.